distributed/scheduler.py

```diff
@@ -4368,23 +4368,40 @@ def update_graph(
         if isinstance(annotations, ToPickle):  # type: ignore
             # FIXME: what the heck?
             annotations = annotations.data  # type: ignore
+
         stimulus_id = stimulus_id or f"update-graph-{time()}"
         (
             dsk,
             dependencies,
-            layer_annotations,
-            pre_stringified_keys,
-        ) = self.materialize_graph(graph)
+            annotations_by_type,
+        ) = self.materialize_graph(graph, annotations)
+
+        if internal_priority is None:
+            # Removing all non-local keys before calling order()
+            dsk_keys = set(dsk)  # intersection() of sets is much faster than dict_keys
+            stripped_deps = {
+                k: v.intersection(dsk_keys)
+                for k, v in dependencies.items()
+                if k in dsk_keys
+            }
+            internal_priority = dask.order.order(dsk, dependencies=stripped_deps)
```

Comment:
this is just a preparatory step. In the follow up PR I will rename the variables. This is just about the logical grouping of doing static/non-mutating operations first instead of grouping the operations by content